src/moj/components/date-picker/date-picker.js

```diff
@@ -0,0 +1,596 @@
+
+    let disabled
+
+    if (thisDay < this.minDate) {
+      disabled = true
+    }
+    if (thisDay > this.maxDate) {
+      disabled = true
+    }
+
+    this.calendarDays[i].update(thisDay, hidden, disabled)
+
+    thisDay.setDate(thisDay.getDate() + 1)
+  }
+}
+
+Datepicker.prototype.setCurrentDate = function (focus = true) {
+  const { currentDate } = this
+
+  this.calendarDays.forEach(calendarDay => {
+    calendarDay.button.setAttribute('tabindex', -1)
+    calendarDay.button.classList.remove('moj-datepicker-selected')
+    const calendarDayDate = calendarDay.date
+    calendarDayDate.setHours(0, 0, 0, 0)
+
+    const today = new Date()
+    today.setHours(0, 0, 0, 0)
+
+    if (calendarDayDate.getTime() === currentDate.getTime() && !calendarDay.disabled) {
+      if (focus) {
+        calendarDay.button.setAttribute('tabindex', 0)
+        calendarDay.button.focus()
+        calendarDay.button.classList.add('moj-datepicker-selected')
+      }
+    }
+
+    if (this.inputDate && calendarDayDate.getTime() === this.inputDate.getTime()) {
+      calendarDay.button.classList.add('moj-datepicker__current')
+      calendarDay.button.setAttribute('aria-selected', true)
+    } else {
+      calendarDay.button.classList.remove('moj-datepicker__current')
+      calendarDay.button.removeAttribute('aria-selected')
+    }
+
+    if (calendarDayDate.getTime() === today.getTime()) {
+      calendarDay.button.classList.add('moj-datepicker__today')
+    } else {
+      calendarDay.button.classList.remove('moj-datepicker__today')
+    }
+  })
+
+  // if no date is tab-able, make the first non-disabled date tab-able
+  if (!focus) {
+    const enabledDays = this.calendarDays.filter(calendarDay => {
+      return window.getComputedStyle(calendarDay.button).display === 'block' && !calendarDay.button.disabled
+    })
+
+    enabledDays[0].button.setAttribute('tabindex', 0)
+
+    this.currentDate = enabledDays[0].date
+  }
+}
+
+Datepicker.prototype.selectDate = function (date) {
+  this.$calendarButton.querySelector('span').innerText = `Choose date. Selected date is ${this.formattedDateHuman(
+    date,
+  )}`
+  this.$input.value = this.formattedDateFromDate(date)
+
+  const changeEvent = new Event('change', { bubbles: true, cancelable: true })
+  this.$input.dispatchEvent(changeEvent)
+
+  this.closeDialog()
+}
+
+Datepicker.prototype.isOpen = function () {
+  return this.dialogElement.classList.contains('moj-datepicker__dialog--open')
+}
+
+Datepicker.prototype.toggleDialog = function (event) {
+  event.preventDefault()
+  if (this.isOpen()) {
+    this.closeDialog()
+  } else {
+    this.setMinAndMaxDatesOnCalendar()
+    this.openDialog()
+  }
+}
+
+Datepicker.prototype.openDialog = function () {
+  // display the dialog
+  this.dialogElement.style.display = 'block'
+  this.dialogElement.classList.add('moj-datepicker__dialog--open')
+
+  // position the dialog
+  this.dialogElement.style.left = `${this.$input.offsetWidth + 16}px`
```

Comment:
If the button is on the right-hand side of the window, this overflows and stretches the page width (or is inaccessible if in an unscrollable container)